base/dcp_sharded_cfg.go

```diff
@@ -0,0 +1,146 @@
+package base
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/couchbase/cbgt"
+	"github.com/couchbase/gocb"
+)
+
+// CfgSG is an implementation of Cfg that uses Sync Gateway's existing
+// bucket connection, and existing caching feed to get change notifications.
+//
+type CfgSG struct {
+	bucket        *gocb.Bucket
+	loggingCtx    context.Context
+	subscriptions map[string][]chan<- cbgt.CfgEvent // Keyed by key
+	lock          sync.Mutex                        // mutex for subscriptions
+}
+
+// NewCfgCB returns a Cfg implementation that reads/writes its entries
+// from/to a couchbase bucket, using DCP streams to subscribe to
+// changes.
+//
+// urlStr: single URL or multiple URLs delimited by ';'
+// bucket: couchbase bucket name
+func NewCfgSG(bucket *gocb.Bucket) (*CfgSG, error) {
+
+	cfgContextID := fmt.Sprintf("%s-%s", MD(bucket.Name()), DCPImportFeedID)
```

Comment:
While we're changing this, could switch to the cheaper string concatenation instead.

```suggestion
	cfgContextID := MD(bucket.Name()).Redact() + "-" + DCPImportFeedID
```

Reply:
Makes sense - will standardize into a helper function (this is being done in a few places) and make those changes.